### PR TITLE
chore: migrate secrets to vault

### PR DESCRIPTION
**Pull request by narikazu**
# Summary

Prepare to use Vault. Playbook is [here](https://www.notion.so/artsy/Migration-of-Artsy-Projects-Secrets-07e5ce9e1ed044dba8d4b7c3f0ae1ccf?pvs=4#3b8a63a3e93548d8a0f435e8b57309ee)

# Ticket
https://artsyproduct.atlassian.net/browse/PHIRE-802

### CODE REVIEW

File: Dockerfile
Comment (ovasdi) [inline Dockerfile:1]: Looks like the Dockerfile is missing the `ENTRYPOINT ["/usr/bin/dumb-init", "./load_secrets_and_run.sh"]`

File: hokusai/production.yml
Comment (ovasdi) [inline hokusai/production.yml:1]: Seems `volumeMounts:` is missing in this file.

**Review by ovasdi — COMMENTED** [review]
Looks good mostly. Just a couple of things appear to be missing: entrypoint and volumeMounts in prod spec

File: Dockerfile
Comment (artsyjian) [inline Dockerfile:16]: i think this COPY is not required. the [ADD](https://github.com/artsy/aprd/blob/c6abf0508d35bac1877ddaafff6afbdd618485af/Dockerfile#L8) takes care of copying `scripts` dir. this Dockerfile has only one build stage, so all the files added by that ADD are kept in the final image. MP was different in that it has 2 stages ([#1](https://github.com/artsy/metaphysics/blob/d0d6ab07696d58582624b05eebb8d7e525e61e46/Dockerfile#L4), [#2](https://github.com/artsy/metaphysics/blob/d0d6ab07696d58582624b05eebb8d7e525e61e46/Dockerfile#L36)), and stage #2 is the final image and it has only those files that are explicitly [copied](https://github.com/artsy/metaphysics/blob/d0d6ab07696d58582624b05eebb8d7e525e61e46/Dockerfile#L51) from stage #1, therefore, the `COPY` is required there.

File: hokusai/production.yml
Comment (narikazu) [reply hokusai/production.yml:1]: Thanks! Added `valumeMounts` in this commit:  https://github.com/artsy/aprd/pull/482/commits/4590b73d2104f886b72b2eba5ef08eb1a0db495b

File: Dockerfile
Comment (narikazu) [reply Dockerfile:1]: Thank you for the feedback! Added the `ENTRYPOINT` in this commit: https://github.com/artsy/aprd/pull/482/commits/204a8ff1c1c260ac71fe28c05a0113d82acee874

File: Dockerfile
Comment (narikazu) [reply Dockerfile:16]: Oh, I didn't know that we don't need the `ADD` command. I've removed it form the Dockerfile.

**Comment by narikazu** [comment]
@artsyjian  @ovasdi I've addressed all of the comments you guys left. Please have a look at the PR again 🙏

File: Dockerfile
Comment (ovasdi) [reply Dockerfile:1]: @narikazu Sorry I didn't realize that aprd does not include dumb-init, like [MP does](https://github.com/artsy/metaphysics/blob/main/Dockerfile#L11). I wonder if that should be added to the Dockerfile 🤔 
Looks like the [latest CI failure](https://app.circleci.com/pipelines/github/artsy/aprd/2468/workflows/4a8626b4-7a15-4360-8350-be457b77e236/jobs/2025?invite=true#step-104-31783_225) is because of dumb-init script missing

Comment (ovasdi) [reply Dockerfile:1]: @artsyjian do you see any issue with just installing `dumb-init` the way its done in MP?

Comment (artsyjian) [reply Dockerfile:1]: `dumb-init` is our standard. we could add it. but if there's a reason that it's been omitted in this repo, i am not sure we want to spend time on that. i'll leave it up to @narikazu.

File: Dockerfile
Comment (artsyjian) [inline Dockerfile:15]: i think this has to be right above [CMD](https://github.com/artsy/aprd/blob/c6abf0508d35bac1877ddaafff6afbdd618485af/Dockerfile#L32).

**Review by artsyjian — APPROVED** [review]
Looks good. Just one comment on ENTRYPOINT.

File: Dockerfile
Comment (narikazu) [reply Dockerfile]: Oh, i didn't know that. Updated.

File: Dockerfile
Comment (narikazu) [reply Dockerfile:1]: Sure! Removed `dumb-init` from `ENTRYPOINT`.

Comment (artsyjian) [reply Dockerfile:1]: > Oh OK. So if dumb-init is not required then entrypoint could just be ENTRYPOINT ["./load_secrets_and_run.sh"]?

yep.

Comment (ovasdi) [reply Dockerfile:1]: Cool. One more correction on my suggestion then, following [this change](https://github.com/artsy/aprd/pull/482/commits/df386b71a6c68557f20f36f15850b2af3b067f5b), entrypoint needs to point to `scripts/`.  `ENTRYPOINT ["./scripts/load_secrets_and_run.sh"]`, i think. 😅

Comment (narikazu) [reply Dockerfile:1]: OK, the CI is still failing. I'll look into it tomorrow.

Comment (artsyjian) [reply Dockerfile:1]: Sure. May just have to [add execution permission](https://www.notion.so/artsy/Migration-of-Artsy-Projects-Secrets-07e5ce9e1ed044dba8d4b7c3f0ae1ccf?pvs=4#13fcab0764a080239e7bec8198f13e95) to the file:
```
artsy-2:aprd jxu$ ls -l scripts/
total 8
-rw-r--r--  1 jxu  staff  184 Nov 20 07:16 load_secrets_and_run.sh
```

**Comment by narikazu** [comment]
I'll be off this afternoon, so if something goes wrong after merging this PR, I can't fix it today. I'll merge this branch and migrate secrets tomorrow.

**Pull request merged**
